src/NHibernate/Collection/Generic/PersistentGenericBag.cs

```diff
@@ -295,12 +295,15 @@ public override bool EqualsSnapshot(ICollectionPersister persister)
 				return false;
 			}
 
-			foreach (var elt in _gbag)
+			for (var i = 0; i < _gbag.Count; i++)
 			{
-				if (CountOccurrences(elt, _gbag, elementType) != CountOccurrences(elt, sn, elementType))
-				{
+				if(elementType.IsSame(_gbag[i], sn[i]))
```

Comment:
No need to count elements placed on the same positions.

Reply:
New white space glitch after if.